services/course-material/src/components/Page.tsx

```diff
@@ -26,6 +27,31 @@ const Page: React.FC<Props> = ({ onRefresh }) => {
   const courseId = pageContext?.pageData?.course_id
   const pageId = pageContext?.pageData?.id
 
+  const [selectedBlockId, setSelectedBlockId] = useState(document.activeElement?.id ?? null)
```

Comment:
Custom hook useSelectedBlockId that returns [selectedBlockId, clearSelectedBlockId]